mlir/lib/IR/AffineExpr.cpp

```diff
@@ -237,6 +239,144 @@ unsigned AffineDimExpr::getPosition() const {
   return static_cast<ImplType *>(expr)->position;
 }
 
+/// Returns the divide type which are as follow floorrdiv = 1, ceildiv = 2,
+/// and otherwise 0.
```

Comment:
You can't introduce such hardcoded mappings! (1, 2, etc.). Drop this function and just use AffineExprKind at the call site.